File: my-folder/0409-longest-palindrome/solution.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        letter_dict = {}
        for i in s:
            if i in letter_dict:
                letter_dict[i]+=1
            else:
                letter_dict[i]=1
        count = 0
        flag = 0
        for i in letter_dict:
            count+=letter_dict[i]//2
            if letter_dict[i]%2==1:
                flag = 1
        return count*2+flag
        

File: my-folder/0409-longest-palindrome/solution.java
class Solution {
    public int longestPalindrome(String s) {
      // s should be a dict for 
      int[] dict = new int[256];
      for (char c : s.toCharArray()) {
        dict[c-'A']++;
      }
      
      int maxLen = 0;
      
      for (int count : dict) {
        if (count % 2 == 0) {
          maxLen += count;
        } else {
          // if the occurrence is even, then count-1 would be 
          maxLen += count-1;
        }
      }
      
      for (int count : dict) {
        if (count %2 != 0) {
          return maxLen+1;
        }
      }
      return maxLen;
    }
}
